chart/templates/webserver/webserver-ingress.yaml

```diff
@@ -39,31 +39,38 @@ spec:
   {{- if .Values.ingress.web.tls.enabled }}
   tls:
     - hosts:
-        - {{ .Values.ingress.web.host }}
+{{- if .Values.ingress.web.tls.enabled }}
+        {{- range .Values.ingress.web.hosts | default (list .Values.ingress.web.host) }}
+        - {{ . | quote }}
+        {{- end }}
+{{- end }}
       secretName: {{ .Values.ingress.web.tls.secretName }}
   {{- end }}
   rules:
+    {{- range .Values.ingress.web.hosts | default (list .Values.ingress.web.host) }}
     - http:
         paths:
-          {{- range .Values.ingress.web.precedingPaths }}
+          {{- range $.Values.ingress.web.precedingPaths }}
           - path: {{ .path }}
             backend:
               serviceName: {{ .serviceName }}
               servicePort: {{ .servicePort }}
           {{- end }}
           - backend:
-              serviceName: {{ .Release.Name }}-webserver
+              serviceName: {{ $.Release.Name }}-webserver
               servicePort: airflow-ui
-{{- if .Values.ingress.web.path }}
-            path: {{ .Values.ingress.web.path }}
+{{- if $.Values.ingress.web.path }}
+            path: {{ $.Values.ingress.web.path }}
 {{- end }}
-          {{- range .Values.ingress.web.succeedingPaths }}
+          {{- range $.Values.ingress.web.succeedingPaths }}
           - path: {{ .path }}
             backend:
               serviceName: {{ .serviceName }}
               servicePort: {{ .servicePort }}
           {{- end }}
-{{- if .Values.ingress.web.host }}
-      host: {{ .Values.ingress.web.host }}
+
+{{- if . }}
+      host: {{ . | quote }}
 {{- end }}
```

Comment:
```suggestion
      host: {{ . | quote }}
```

Not needed, right?

Reply:
So this conditional was already present in the previous iteration of the chart.  Reading the ingress docs, looks like setting no host matches anything without a HTTP host header.  Now I am not sure if no host entry is the same as an empty host entry.

Reply:
I just mean with `range`, I don't think `.` can ever be falsy.

However, that is a good callout that with this change you can't set up the backend without having a host, which would be a breaking change, right?

Reply:
Yes it would be.  The k8s ingress docs say without a host specified the rule matches anything trying to hit the ingress via an IP address ie no host http header.  Now I know this definitely doesn't work on the ingress nginx controller, but no clue about the other ingress controllers.  

However, in running `helm template` for a few scenarios, wrapping this conditional around the host entry does make a difference.  If just `.Values.web.ingress.enabled` is set to `true` and everything else is kept as the defaults, then the host entry will NOT show up with the conditional present, but if that conditional is removed, then host will should up as an empty string ie the default value.  How that's dealt with seems to depend on the ingress controller.

I am of the mind to leave this as is, unless there is some other data/docs that says the above difference doesn't matter.

Reply:
@jedcunningham added a test case for each ingress to cover the above. Let me know what you think.

Reply:
Now I follow, I wasn't groking `list("")` still runs through with `. := ""` 👍.